doc/tutorials/debug.rst

```diff
@@ -0,0 +1,322 @@
+.. _acrn-debug:
+
+ACRN Debugging Tools
+####################
+
+This document describes how to use ACRN tools to collect log
+and trace data for debugging.
+
+ACRN Console Command
+********************
+
+The ACRN console provides shell commands for a user to check system states
+and environment settings. See the :ref:`acrnshell` documentation for a
+full list of commands, or see a summary of available commands by using
+the ``help`` command within the ACRN shell.
+
+
+An example
+**********
+
+As an example, we'll show how to obtain the interrupts of a pass-through USB device.
+
+First, we can get the USB controller BDF number (0:15.0) through the
+following command in the SOS console::
+
+   lspci | grep "USB controller"
+
+
+.. figure:: images/debug_image19.png
+   :align: center
+
+   USB controller BDF information
+
+Second, we use the ``pt`` command in the ACRN console, and use this BDF number
+to find the interrupt vector (VEC) "0x31".
+
+.. figure:: images/debug_image20.png
+   :align: center
+
+   USB controller interrupt information
+
+Finally we use the ``int`` command in the ACRN console, and use this
+interrupt vector to find the interrupt number (909) on CPU3.
+
+.. figure:: images/debug_image21.png
+   :align: center
+
+   USB controller interrupt number information
+
+ACRN Log
+********
+
+ACRN log provides console log and mem log for a user to analyze.
+We can use console log to debug directly, while mem log is a userland tool
+used to capture a ACRN hypervisor log. NPK log contains userspace and
+kernel log, SOS userspace/kernel log, and hypervisor log.
+
+Turn on the logging info
+========================
+
+ACRN enables a console log by default.
+
+For a mem log, use this command in an SOS console, and then reboot the
+SOS::
+
+   $ systemctl enable acrnlog
+   $ reboot
+
+For NPK log, we need to enable NPK in the SOS, Hypervisor, and Android
+(AaaG).
```

Comment:
Is NPK specific to Android? (we do not have any information on the website on how to set-up AaaG)

Reply:
Through my research, i find that NPK isn't an open-source tool, and the log analysis tool is only for OEM user, so need we add this section into document?

Reply:
Suggest to remove NPK part for now since it is not for opensource yet.